### ISSUE TITLE
skill book enchants upgrade with weapon/armor (bug?)

### ISSUE THREAD

**Issue by ImperialUser** [Heishe/idleraiders-second-run#437]
Just recently, skill books that get enchanted will have their enchantments upgrade when upgrading weapon/armor.  Older skill book enchantments are still the same level, it's just skill books that get enchanted now that upgrade.

Info: My books with 25% stay at 25% when upgrading weapon/armor research, but the skills books at 30% have improved to 31% and 32% when I upgraded my weapon/armor research without reenchanting them.

**Comment by TravisGesslein** [comment]
The intended behavior is that only skillbooks that are enchanted after you unlocked the weapon and armor improvement research gain the improved bonuses. Enchantments that were applied before should stay at the same value. 

But it sounds like some of your skillbooks with existing enchantments have their values upgraded, too? Is that what's happening?

**Comment by ImperialUser** [comment]
At lvl 25 research I upgraded all my skill books to lvl 25 enchants.
My skill books have always stayed at their enchanted level up to this point. So as I go 26 research, 27 research, 28 research those books stay at lvl 25 enchant.

At lvl 30 research, a few days ago, I upgraded a handful of important skillbooks to lvl 30 enchant.
As I've gained research, lvl 31, lvl 32, my skill books that I enchanted to lvl 30 are now upgrading along with my research lvl without reenchanting them, but the books I previously enchanted at lvl 25 are still lvl 25.

**Comment by TravisGesslein** [comment]
Yes, that's intended (also it's always worked this way, at least we've changed nothing about it)!

Serves multiple purposes. For one thing, from a technical perspective it's actually not easy to track all skillbooks that have gotten enchants at some levels and then upgrade them to the correct level. It's doable, it's just that the current code doesn't support it. Second, it's a nice gold and crafting resource sink, to make those currencies at least a bit valuable (though that said, more will come!).

**Comment by ImperialUser** [comment]
So which one is correct, because both are happening for me.

Are you supposed to re-enchant equipment to have it upgrade to your current research lvl?
Or do enchantment books upgrade automatically to the new value when doing the research?

The skill books I've enchanted in the past have always stayed at their enchantment level after doing research.  Recently the books that I upgraded to lvl 30 have been upgrading without re-enchanting them.

Right now I have skill books at lvl 25 and lvl 32.  When I get lvl 33 research I'll have skill books at lvl 25 and lvl 33.

Hopefully that's clearer?

**Comment by ImperialUser** [comment]
PS: While we are on the subject of enchants, a cap of 100% should be in place for reduction enchants like cooldown and cost.  What happens if you get research 101 and your cooldown is -101%?

**Comment by TravisGesslein** [comment]
Oh, yes that's clearer. Intended would be your first option:

> Are you supposed to re-enchant equipment to have it upgrade to your current research lvl?

Can you easily reproduce this?

Regarding caps: That's true. In general, a lot of upgrades in this game probably need to be capped. It creates a problem where it's impossible to balance bosses properly because, taking bulwark and cooldown reduction (or buff duration) as an example, a lot of players who have figured it out run around perma-bulwarked would have a much too easy time with anything that is designed without perma-bulwark in mind. on the other hand it's really cool that you can reach that kind of breakpoint, where your tank becomes 'something else'.

not sure how to deal with this yet (open for suggestions, btw). since that kind of balancing mostly goes completely crazy once you've been playing a very long time (probably long after you've beaten the last boss), it's probably subject to change with the first post-release patch that will bring a new raid dungeon.

**Comment by ImperialUser** [comment]
I can let you know after I get lvl 33 if it happens again, idk what the trigger cause was though.  Need a lot of obsidian for these later levels. Just books that I upgrade at lvl 30 are now upgrading when I do research.  But since it's not happening to all of my skill books, just the ones I upgraded at lvl 30, I'd look at the items themselves.

If it's a combination issue, can also verify it didn't happen at lvl 25. So possibly somewhere between lvl 25 and 30 item enchants turns the item from a "skillbook" to an "equipment"?

I'll also try re-enchanting a book I got at level 25, 30 and a new one.  See if they all upgrade when I get 33.

**Comment by ImperialUser** [comment]
Lvl 32 research skillbook info:

+25% chance cleave (old)
+32% chance cleave (old lvl 25, renchanted to 32)
-32% cost warcray (lvl 30 that was at lvl 32, renchanted to lvl 32)
+32% damage mighty blow (lvl 30 at lvl 32)
+160% duration mark enemy (new book newly enchanted at lvl 32)

--research lvl 33--
25% chance cleave (old) stayed at lvl 25
All other books upgraded to lvl 33.

**Comment by ImperialUser** [comment]
Whatever happened, seems to have fixed itself.  My game broke and went back to research lvl 32, when I got lvl 33 my stuff stayed at lvl 32 this time.

**Comment by ImperialUser** [comment]
Whatever happened, is happening again.  My old skill books at 25% are still 25%, but now all my other skill books are at 37% without reenchanting them.

Also just to clear a previous error:
When I was talking about lvl 25 and lvl 30 etc in previous comments, those are actually -10.  I was thinking lvl 25 because 25%.  My old skill books are actually lvl 15 (25%) as an example.  Just for clarification.
